titiler/endpoints/factory.py

```diff
@@ -677,6 +680,225 @@ def part(
             return Response(content, media_type=format.mimetype, headers=headers)
 
 
+@dataclass
+class MultiBaseTilerFactory(TilerFactory):
+    """Custom Tiler Factory for MultiBaseReader classes.
+
+    Note:
+        To be able to use the rio_tiler.io.MultiBaseReader we need to be able to pass a `assets`
+        argument to most of its methods. By using the `AssetsBidxExprParams` for the `layer_dependency`, the
+        .tile(), .point(), .preview() and the .part() methods will receive assets, expression or indexes arguments.
+
+        The rio_tiler.io.MultiBaseReader  `.info()` and `.metadata()` have `assets` as
+        a requirement arguments (https://github.com/cogeotiff/rio-tiler/blob/master/rio_tiler/io/base.py#L365).
+        This means we have to update the /info and /metadata endpoints in order to add the `assets` dependency.
+
+    """
+
+    reader: Type[MultiBaseReader] = field()
+
+    # Assets/Indexes/Expression Dependencies
+    layer_dependency: default_deps_type = AssetsBidxExprParams
+
+    # Overwrite the `/info` endpoint to return the list of assets when no assets is passed.
+    def info(self):
+        """Register /info endpoint."""
+
+        @self.router.get(
+            "/info",
+            response_model=Union[List[str], Dict[str, Info]],
+            response_model_exclude={"minzoom", "maxzoom", "center"},
+            response_model_exclude_none=True,
+            responses={
+                200: {
+                    "description": "Return dataset's basic info or the list of available assets."
+                }
+            },
+        )
+        def info(
+            src_path=Depends(self.path_dependency),
+            asset_params=Depends(AssetsBidxParams),
+            kwargs: Dict = Depends(self.additional_dependency),
+        ):
+            """Return dataset's basic info or the list of available assets."""
+            with rasterio.Env(**self.gdal_config):
+                with self.reader(src_path.url, **self.reader_options) as src_dst:
+                    if not asset_params.assets:
+                        return src_dst.assets
```

Comment:
Here we could also choose to raise an error telling `assets` is missing from the querystring 🤷‍♂️